### DISCUSSION TITLE
Livewire.* methods always return empty after livewire:navigated

### DISCUSSION THREAD

**Discussion by ianlandsman** [livewire/livewire#7130] [1. Help] [answered]
If you put this in your app.js

`document.addEventListener('livewire:navigated', () => {
    console.log(Livewire.all())
})`

You'll get the array of components on the initial page load, but every subsequent wire:navigate it will log an empty array even though there are components and they seem to be rerendering as they get new wire IDs. 

It seems like the Livewire.* methods should return the components of the current page after a wire:navigate. Is that not correct? Perhaps I'm misunderstanding how those methods work. I don't see anything in the docs that would suggest they shouldn't work.

**Comment by joshhanley** [comment, marked as answer]
We're in the process of investigating all wire:navigate issues and I have tested out your scenario. I believe this has been fixed so I'm going to close this. If you're still experiencing problems, please open a new discussion. Thanks!
